kube-runtime/src/controller/mod.rs

```diff
@@ -758,6 +758,22 @@ where
         self
     }
 
+    #[must_use]
+    pub fn reconcile_on(
```

Comment:
Needs a no_run doc test example. your example in the PR is probably fine if you drop the pod_mapper part.